pkg/inject/inject.go

```diff
@@ -774,73 +783,73 @@ func (conf *ResourceConfig) applyAnnotationOverrides(values *l5dcharts.Values) {
 		}
 	}
 
-	if override, ok := conf.getOverride(k8s.ProxyCPURequestAnnotation); ok {
-		res, err := k8sResource.ParseQuantity(override)
+	if override, ok := annotations[k8s.ProxyCPURequestAnnotation]; ok {
+		_, err := k8sResource.ParseQuantity(override)
 		if err != nil {
 			log.Warnf("%s (%s)", err, k8s.ProxyCPURequestAnnotation)
 		} else {
-			values.Global.Proxy.Resources.CPU.Request = res.String()
+			values.Global.Proxy.Resources.CPU.Request = override
 		}
 	}
 
-	if override, ok := conf.getOverride(k8s.ProxyMemoryRequestAnnotation); ok {
-		res, err := k8sResource.ParseQuantity(override)
+	if override, ok := annotations[k8s.ProxyMemoryRequestAnnotation]; ok {
+		_, err := k8sResource.ParseQuantity(override)
 		if err != nil {
 			log.Warnf("%s (%s)", err, k8s.ProxyMemoryRequestAnnotation)
 		} else {
-			values.Global.Proxy.Resources.Memory.Request = res.String()
+			values.Global.Proxy.Resources.Memory.Request = override
 		}
 	}
 
-	if override, ok := conf.getOverride(k8s.ProxyCPULimitAnnotation); ok {
-		res, err := k8sResource.ParseQuantity(override)
+	if override, ok := annotations[k8s.ProxyCPULimitAnnotation]; ok {
+		_, err := k8sResource.ParseQuantity(override)
 		if err != nil {
 			log.Warnf("%s (%s)", err, k8s.ProxyCPULimitAnnotation)
 		} else {
-			values.Global.Proxy.Resources.CPU.Limit = res.String()
+			values.Global.Proxy.Resources.CPU.Limit = override
 		}
 	}
 
-	if override, ok := conf.getOverride(k8s.ProxyMemoryLimitAnnotation); ok {
-		res, err := k8sResource.ParseQuantity(override)
+	if override, ok := annotations[k8s.ProxyMemoryLimitAnnotation]; ok {
```

Comment:
Looks Good. :+1: Just to confirm, This is possible because

- Flags are directly overridden on top of default Values in `applySetFlags`
- `nsAnnotations` are overridden with `pod.meta.annoations`.

So, We have a single `annotations` struct consisting of final overrides?

Reply:
Yup!